src/utilities/validators/base-validator.ts

```diff
@@ -224,6 +225,17 @@ abstract class BaseValidator {
       }
     });
 
+    const missingOptionalProperties = expectedProperties.filter(
+      (property) => !actualProperties.includes(property),
+    );
```

Comment:
This property array will also include required properties right? Is that intended?

Reply:
my thought was that it didn't matter since they'd get the hard failure for the required properties, but the wording of the warning does specify `non required` so you're right i should update this